x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/LuceneQueryEvaluator.java

```diff
@@ -54,6 +54,7 @@ public record ShardConfig(Query query, IndexSearcher searcher) {}
     private final List<ShardState> perShardState;
 
     protected LuceneQueryEvaluator(BlockFactory blockFactory, ShardConfig[] shards) {
+        assert shards.length > 0 : "LuceneQueryEvaluator expects shard configs";
```

Comment:
This will help detect non-pushable expressions that need to be created with ShardConfig information